### PR TITLE
:zap: change fastqc tool to take parameter file as input

### PR DESCRIPTION
**Pull request by sickler-alex**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Update fastqc tool to take the parameter file as input.

Fixes # 866 (partially)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] [Cavatica test run](https://cavatica.sbgenomics.com/u/d3b-bixu/dev-sample-qc/tasks/283ae9d5-dd83-426b-97ca-4809456f340f/#)
- [X] [Updated test](https://cavatica.sbgenomics.com/u/d3b-bixu/dev-sample-qc/tasks/cceeae64-9db3-4ec1-a178-cfa7ddf50af0/)
- [X] [Test with no input param file](https://cavatica.sbgenomics.com/u/d3b-bixu/dev-sample-qc/tasks/f630bd54-e5f0-4fc1-b5bd-f82e60fce685/)

**Test Configuration**:
* Environment: Cavatica
* Test files:

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: tools/fastqc.cwl [hunk 1]
```diff
@@ -11,61 +11,21 @@ requirements:
...
     valueFrom: >-
-     -l fastqc_params -t ${return Math.min(inputs.sequences.length, inputs.max_cpu)} -o .
+     -l $(inputs.fastqc_params.path) -t ${return Math.min(inputs.sequences.length, inputs.max_cpu)} -o .
 
 inputs:
   sequences: {type: 'File[]', inputBinding: { position: 99 }, doc: "set of sequences being run can be either fastqs or bams"}
   return_raw_data: {type: boolean?, doc: "TRUE: return zipped raw data folder or FALSE: only return summary HTML"}
   ram: {type: ['null', int], default: 2, doc: "In GB"}
   max_cpu: {type: ['null', int], default: 8, doc: "Maximum number of CPUs to request"}
+  fastqc_params: {type: 'File', doc: "fastqc parameter file to use"}
```
Comment (dmiller15) [inline tools/fastqc.cwl:28]: This should be optional so users can just run with the default that exists in the repo: `/FastQC/Configuration/limits.txt`

**Review by dmiller15 — APPROVED** [review]

File: tools/fastqc.cwl [hunk 2]
```diff
@@ -18,14 +18,14 @@ arguments:
...
-  sequences: {type: 'File[]', inputBinding: { position: 99 }, doc: "set of sequences being run can be either fastqs or bams"}
+  sequences: {type: 'File[]', inputBinding: {position: 99}, doc: "set of sequences being run can be either fastqs or bams"}
   return_raw_data: {type: boolean?, doc: "TRUE: return zipped raw data folder or FALSE: only return summary HTML"}
   ram: {type: ['null', int], default: 2, doc: "In GB"}
   max_cpu: {type: ['null', int], default: 8, doc: "Maximum number of CPUs to request"}
-  fastqc_params: {type: 'File', doc: "fastqc parameter file to use"}
+  fastqc_params: {type: 'File?', inputBinding: {position: 1, prefix: -l}, doc: "fastqc parameter file to use"}
 
 outputs:
   output_summarys:
```
Comment (migbro) [inline tools/fastqc.cwl:31]: Just a spelling thing: `output_summaries`

**Review by migbro — CHANGES_REQUESTED** [review]
Just a minor request to fix and output name typo, I'll approve after that :)

**Review by migbro — APPROVED** [review]
Looks good, thanks!

**Pull request merged**